qiskit/wrapper/defaultqiskitprovider.py

```diff
@@ -156,14 +156,13 @@ def add_provider(self, provider):
         # checks for equality of provider instances, based on the __eq__ method
         if provider not in self.providers:
             self.providers.append(provider)
+            if common_backends:
+                logger.warning(
+                    'The backend names "%s" of this provider are already in use. '
+                    'Refer to documentation for `available_backends()` and `unregister()`.',
+                    list(common_backends))
         else:
-            raise QISKitError("The same provider has already been registered!")
-
-        if common_backends:
-            logger.warning(
-                'The backend names "%s" of this provider are already in use. '
-                'Refer to documentation for `available_backends()` and `unregister()`.',
-                list(common_backends))
+            warnings.warn("Skipping registration: The provider is already registered.")
```

Comment:
Can you return the _previous_ provider if the function enters this `else`, for consistency? While in practice both the provider that is trying to be registered and the already registered should be the same for most purposes and have an equality comparison, it is not thorough, and it feels more suited to return the old one to avoid potential hard to track side effects (ie. with a "peace of mind" bonus).

Also, can you update the docstring for this `add_provider()` method, taking out the `Raises` information?

Reply:
Done.